provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1460,38 +1517,55 @@ func nodeGroupProperties(cluster, v2 bool) map[string]schema.PropertySpec {
 		},
 		"labels": {
 			TypeSpec: schema.TypeSpec{
-				Type:                 "object",
-				AdditionalProperties: &schema.TypeSpec{Type: "string"},
+				Type: "object",
+				AdditionalProperties: &schema.TypeSpec{
+					Type:  "string",
+					Plain: true,
+				},
+				Plain: true,
 			},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/nodegroup.ts#L220-L224